examples/controller/index.js

```diff
@@ -0,0 +1,42 @@
+import 'webcomponents.js'; // polyfill
+
+// import from the same repo. in a different repo you'd use:
+// import { Component, h } from 'panel';
+import {ControlledComponent, StateController, h} from '../../lib';
+
+class CounterController extends StateController {
+  get defaultState() {
+    return {
+      count: 0,
+    };
+  }
+
+  incrCounter() {
+    const count = this.state.count + 1;
+    this._update({count});
+  }
+
+  decrCounter() {
+    const count = this.state.count - 1;
+    this._update({count});
+  }
+
+  getCount() {
+    return this.state.count;
+  }
+}
+
+document.registerElement(`counter-app`, class extends ControlledComponent {
+  get config() {
+    return {
+      controller: new CounterController(),
+      template: () => h(`div.counter`, [
+        h(`div.val`, `Counter: ${this.controller.getCount()}`),
```

Comment:
I like the notion of the StateController as a state-selector layer as well as state-mutator.